packages/mobile/src/harmony-native/components/Button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,169 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { useControlled } from '@audius/harmony/src/hooks/useControlled'
+import type { ReactNativeStyle } from '@emotion/native'
+import { useTheme } from '@emotion/react'
+
+import { IconCloseAlt, Text } from '@audius/harmony-native'
+
+import { BaseButton } from '../BaseButton/BaseButton'
+
+import type { FilterButtonProps } from './types'
+
+export const FilterButton = (props: FilterButtonProps) => {
+  const {
+    value: valueProp,
+    // children,
+    label: labelProp,
+    // onChange,
+    onPress,
+    onOpen,
+    onReset,
+    disabled,
+    variant = 'fillContainer',
+    size = 'default',
+    iconRight
+  } = props
+
+  const { color, cornerRadius, spacing, typography } = useTheme()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const [value, setValue] = useControlled({
+    controlledProp: valueProp,
+    defaultValue: null,
+    stateName: 'value',
+    componentName: 'FilterButton'
+  })
+
+  const [label, setLabel] = useControlled({
```

Comment:
I just did a lil refactor of web FilterButton in my pr, turns out `label` doesn't really need to be controlled. We can change it later though too: https://github.com/AudiusProject/audius-protocol/pull/9004/files#diff-209468f43539e5585a9473fca2d90093bc25db8cf2f3be858c150d67c60beda7L34